analyzers/src/SonarAnalyzer.CSharp/Rules/Utilities/TokenTypeAnalyzer.cs

```diff
@@ -55,42 +47,23 @@ public TokenClassifier(SyntaxToken token, SemanticModel semanticModel, bool skip
             protected override bool IsKeyword(SyntaxToken token) =>
                 SyntaxFacts.IsKeywordKind(token.Kind());
 
-            protected override bool IsRegularComment(SyntaxTrivia trivia)
-            {
-                switch (trivia.Kind())
-                {
-                    case SyntaxKind.SingleLineCommentTrivia:
-                    case SyntaxKind.MultiLineCommentTrivia:
-                        return true;
-
-                    default:
-                        return false;
-                }
-            }
+            protected override bool IsRegularComment(SyntaxTrivia trivia) =>
+                trivia.IsAnyKind(SyntaxKind.SingleLineCommentTrivia, SyntaxKind.MultiLineCommentTrivia);
 
             protected override bool IsNumericLiteral(SyntaxToken token) =>
                 token.IsKind(SyntaxKind.NumericLiteralToken);
 
-            protected override bool IsStringLiteral(SyntaxToken token)
-            {
-                switch (token.Kind())
-                {
-                    case SyntaxKind.StringLiteralToken:
-                    case SyntaxKind.CharacterLiteralToken:
-                    case SyntaxKind.InterpolatedStringStartToken:
-                    case SyntaxKind.InterpolatedVerbatimStringStartToken:
-                    case SyntaxKind.InterpolatedStringTextToken:
-                    case SyntaxKind.InterpolatedStringEndToken:
-                        return true;
-
-                    default:
-                        return false;
-                }
-            }
+            protected override bool IsStringLiteral(SyntaxToken token) =>
+                token.IsAnyKind(
+                    SyntaxKind.StringLiteralToken,
+                    SyntaxKind.CharacterLiteralToken,
+                    SyntaxKind.InterpolatedStringStartToken,
+                    SyntaxKind.InterpolatedVerbatimStringStartToken,
+                    SyntaxKind.InterpolatedStringTextToken,
+                    SyntaxKind.InterpolatedStringEndToken);
 
             protected override bool IsDocComment(SyntaxTrivia trivia) =>
-                trivia.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia) ||
-                trivia.IsKind(SyntaxKind.MultiLineDocumentationCommentTrivia);
+                trivia.IsAnyKind(SyntaxKind.SingleLineDocumentationCommentTrivia, SyntaxKind.MultiLineDocumentationCommentTrivia);
```

Comment:
The is comment and is doc checks are present in multiple rules, maybe they deserve their place in the LanguageFacade and the IsDocComment method will not be neccessary.

Reply:
I'll do that in another PR. It will affect more rules.